nixos/modules/services/networking/wireguard.nix

```diff
@@ -272,22 +280,25 @@ let
         '';
       };
 
-  peerUnitServiceName = interfaceName: publicKey: dynamicRefreshEnabled:
+  peerUnitServiceName = interfaceName: peerName: dynamicRefreshEnabled:
     let
       keyToUnitName = replaceChars
-        [ "/" "-"    " "     "+"     "="      ]
+        [ "/" "-" " " "+" "=" ]
         [ "-" "\\x2d" "\\x20" "\\x2b" "\\x3d" ];
-      unitName = keyToUnitName publicKey;
+      unitName = keyToUnitName peerName;
       refreshSuffix = optionalString dynamicRefreshEnabled "-refresh";
     in
-      "wireguard-${interfaceName}-peer-${unitName}${refreshSuffix}";
+    "wireguard-${interfaceName}-peer-${unitName}${refreshSuffix}";
 
   generatePeerUnit = { interfaceName, interfaceCfg, peer }:
     let
+      peerName = if peer.publicKey != null then peer.publicKey else toString peer.publicKeyFile;
```

Comment:
Using the path to the public key as a peer name can cause conflicts. This also relates to https://github.com/NixOS/nixpkgs/pull/71737 which tried to allow explicitly naming peers, which might be needed to not cause this problem. Ping @delroth @ivan @Lassulus @grahamc